hudi-common/src/main/java/org/apache/hudi/common/table/timeline/TimeGeneratorType.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.table.timeline;
+
+import org.apache.hudi.common.config.EnumDescription;
+import org.apache.hudi.common.config.EnumFieldDescription;
+
+/**
+ * Types of {@link TimeGenerator}.
+ */
+@EnumDescription("Time generator type, indicating the time generator class to use, that implements "
+    + "`org.apache.hudi.common.table.timeline.TimeGenerator`.")
+public enum TimeGeneratorType {
+
+  @EnumFieldDescription("A wait based time generator, holding a mutex lock with skew times to "
+      + "produce globally monotonically increasing timestamp")
+  LOCK_PROVIDER
```

Comment:
rename: WAIT_BASED ? Calling it `LOCK_PROVIDER` is a bit confusing IMO, since we have other meanings for this in the code/docs. 
In fact, call this sth like - `WAIT_TO_ADJUST_SKEW` to make it more explicit?